### DISCUSSION TITLE
Is it possible to mount GRPC endpoints under a subdir?

### DISCUSSION THREAD

**Discussion by vdods** [hyperium/tonic#825] [Q&A]
GRPC uses HTTP, and each RPC maps onto an HTTP endpoint like http://mydomain/endpoints.examples.bookstore.Bookstore/ListShelves (reference https://cloud.google.com/endpoints/docs/grpc/transcoding ).

My question is if/how it's possible to mount an entire GRPC service under a subdirectory (say "fancything"), so that e.g. the above endpoint becomes http://mydomain/fancything/endpoints.examples.bookstore.Bookstore/ListShelves ?

**Comment by davidpdrsn** [comment]
No thats not currently possible.

It might be possible to do with a tower middleware that rewrites the URI if it matches some prefix but its never something I've had to do so not sure.

**Comment by frederikhors** [comment]
@vdods did you find a solution for the prefix?

**Comment by vdods** [comment]
No, I didn't pursue this.
